### PR TITLE
Lp profile

### PR DESCRIPTION
**Pull request by lspdrz**
## Proposed changes

Adds Profile page to Pathways Client

## Types of changes

What types of changes does your code introduce to Appium?

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: src/pages/Profile/index.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import { connect, ConnectedProps } from "react-redux";
+import { fetchCalls } from "src/redux/modules/call";
+import { selectAllCalls } from "src/redux/selectors";
+import { push } from "connected-react-router";
+import "src/i18n/config";
+import Profile from "src/components/Profile";
+
+const mapStateToProps = (state: RootState) => ({
+  user: state.session.user,
+  calls: selectAllCalls(state),
```
Comment (gabrielsaruhashi) [inline src/pages/Profile/index.tsx:12]: moving fwd let's just start using the useselector hook?

I agree that hooks are the future and they reduce the amount of boilerplate code

File: src/pages/Profile/index.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+});
+
+const mapDispatchToProps = { fetchCalls, push };
```
Comment (gabrielsaruhashi) [inline src/pages/Profile/index.tsx:15]: same with use dispathc

File: src/i18n/config.ts [hunk 3]
```diff
@@ -12,6 +13,7 @@ export const resources = {
   en: {
     login: enLogin,
     dashboard: enDashboard,
+    profile: enProfile,
```
Comment (gabrielsaruhashi) [inline src/i18n/config.ts:16]: let's add the es file even if we don't have trnsltion?

File: src/components/menu/Sidebar.tsx [hunk 4]
```diff
@@ -1,21 +1,30 @@
...
 
 export default function Sidebar({
-  navigate,
   isVisible,
   pathname,
+  user,
 }: Props): ReactElement {
-  const [collapsed, setCollapsed] = useState(false);
-
+  const history = useHistory();
```
Comment (gabrielsaruhashi) [inline src/components/menu/Sidebar.tsx:27]: I wouldnt use history directly

We're using connect-react-router to make history changes trackable on the store

I'd move back to passing down push instead and keeping this a dummy component

File: src/components/Profile/ProfileInfo.tsx [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+          style={{
+            height: "60px",
+            backgroundColor: "#F0DEFF",
+            opacity: 0.4,
+          }}
+        />
+        <Space
+          direction="vertical"
+          size="large"
+          style={{ padding: 24, paddingTop: 36 }}
```
Comment (gabrielsaruhashi) [inline src/components/Profile/ProfileInfo.tsx:39]: We shouldnt be using hardcoded values imo
Let's just use tailwind everywhere?

File: src/components/Profile/ProfileInfo.tsx [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+              border: "1px solid #EEEEEE",
+              padding: "25px",
+            }}
+          >
+            <Space style={{ display: "flex", justifyContent: "space-between" }}>
+              <Space>
+                <Avatar
+                  src={user.profileImgPath}
+                  size="large"
+                  style={{
```
Comment (gabrielsaruhashi) [inline src/components/Profile/ProfileInfo.tsx:53]: just use size = {135} instead 

You don't ned to override it on thee style props

I'd also define the avatar sizes on a UI util file 

e.g AVATAR_LARGE = 135

so we can use it here

File: src/components/Profile/ProfileInfo.tsx [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+                icon={<SaveOutlined />}
+                size="large"
+                style={{ borderRadius: 4 }}
+              >
+                {t("profileInfo.editProfile")}
+              </Button>
+            </Space>
+          </PageHeader>
+          <Row>
+            <Col span={15}>
```
Comment (gabrielsaruhashi) [inline src/components/Profile/ProfileInfo.tsx:78]: Should probably be 16 here

Use the gutter props on Row to create spacing

File: src/components/Profile/ProfileInfo.tsx [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+                </p>
+                <p>
+                  <Text>{`${t("profileInfo.birthday")}: ${format(
+                    new Date(user.dateOfBirth),
+                    "MMMM d, yyyy"
+                  )}`}</Text>
+                </p>
+              </Card>
+            </Col>
+            <Col span={1} />
```
Comment (gabrielsaruhashi) [inline src/components/Profile/ProfileInfo.tsx:96]: What's this for?

File: src/components/Profile/ProfileInfo.tsx [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+                      <Text type="secondary">
+                        {t("profileInfo.totalCalls")}
+                      </Text>
+                    </p>
+                    <p>
+                      <Text>{calls.length}</Text>
+                    </p>
+                  </Col>
+                  <Col span={2} />
+                  <Col span={16}>
```
Comment (gabrielsaruhashi) [inline src/components/Profile/ProfileInfo.tsx:111]: use the offset props instead of creating ghost cols

**Review by gabrielsaruhashi — CHANGES_REQUESTED** [review]
some comments + suggestions + nits 🚀

File: src/components/menu/Sidebar.tsx (hunk 4 above)
Comment (lspdrz) [reply src/components/menu/Sidebar.tsx:27]: @gabrielsaruhashi I tried using the push prop before and it wasn't working properly, so I imported history directly. Began working. I'm gonna push the other changes and then try again

File: src/components/Profile/ProfileInfo.tsx (hunk 8 above)
Comment (lspdrz) [reply src/components/Profile/ProfileInfo.tsx:96]: This was to create space betweren the cols. Was unaware of the gutter prop for rows

File: src/App.tsx [hunk 10]
```diff
@@ -21,7 +21,7 @@ const mapStateToProps = (state: RootState) => ({
   session: state.session,
   pathname: state.router.location.pathname,
 });
-const mapDispatchToProps = { fetchConnections, push };
+const mapDispatchToProps = { fetchConnections };
```
Comment (gabrielsaruhashi) [inline src/App.tsx:24]: did push work for you?

File: src/components/menu/Sidebar.tsx (hunk 4 above)
Comment (gabrielsaruhashi) [reply src/components/menu/Sidebar.tsx:25]: oh hmmm it was working for me! 
I use it for the DOC client as well

File: src/components/menu/Sidebar.tsx [hunk 11]
```diff
@@ -25,27 +32,52 @@ export default function Sidebar({
...
-          onClick={() => navigate("home")}
+          <Space>
+            <Avatar src={user.profileImgPath} size="large" />
+            <Typography.Text>{genFullName(user)}</Typography.Text>
+          </Space>
+        </Row>
+        <Menu
+          mode="inline"
+          defaultSelectedKeys={["home"]}
+          style={{ width: "100%" }}
```
Comment (gabrielsaruhashi) [inline src/components/menu/Sidebar.tsx:55]: tailwind?

File: src/components/menu/Sidebar.tsx [hunk 12]
```diff
@@ -25,27 +32,52 @@ export default function Sidebar({
...
+      style={{ boxShadow: "2px 0px 8px rgba(0, 0, 0, 0.15)" }}
     >
-      <div className="logo" />
-      <Menu mode="inline" defaultSelectedKeys={["home"]}>
-        <Menu.Item
-          key="home"
-          icon={<HomeOutlined />}
-          onClick={() => navigate("home")}
+      <Space direction="vertical" style={{ width: "100%" }}>
+        <Row style={{ paddingTop: 8 }}>
```
Comment (gabrielsaruhashi) [inline src/components/menu/Sidebar.tsx:38]: tailwind?

File: src/components/menu/Sidebar.tsx [hunk 13]
```diff
@@ -25,27 +32,52 @@ export default function Sidebar({
   return (
     <Sider
       theme="light"
-      collapsible
-      collapsed={collapsed}
-      onCollapse={() => setCollapsed((collapsed) => !collapsed)}
+      style={{ boxShadow: "2px 0px 8px rgba(0, 0, 0, 0.15)" }}
```
Comment (gabrielsaruhashi) [inline src/components/menu/Sidebar.tsx:35]: hmmm is this on the design?

I'd push to keep the component lib as it is unless we have a good use case

File: src/components/Profile/ProfileInfo.tsx [hunk 14]
```diff
@@ -0,0 +1,111 @@
...
+    .reduce((a, b) => a + b);
+  return (
+    <Layout style={{ minHeight: "100vh", backgroundColor: "#FFFFFF" }}>
+      <Content>
+        <div
+          className="h-16 opacity-40"
+          style={{ backgroundColor: "#F0DEFF" }}
+        />
+        <Space direction="vertical" size="large" className="w-100 p-6 pt-9">
+          <PageHeader className="p-6 border border-solid border-opacity-1 border-gray-200">
```
Comment (gabrielsaruhashi) [inline src/components/Profile/ProfileInfo.tsx:36]: really cool!

File: src/components/Profile/ProfileInfo.tsx [hunk 15]
```diff
@@ -0,0 +1,111 @@
+import { SaveOutlined } from "@ant-design/icons";
+import { Space, PageHeader, Button, Row, Card, Typography, Col } from "antd";
+import Avatar from "antd/lib/avatar/avatar";
+import Layout, { Content } from "antd/lib/layout/layout";
```
Comment (gabrielsaruhashi) [inline src/components/Profile/ProfileInfo.tsx:4]: import {layout} from antd

and then you can do Layout.Content

or const {Content} = Layout

File: src/components/Profile/ProfileInfo.tsx [hunk 16]
```diff
@@ -0,0 +1,111 @@
...
+import Title from "antd/lib/typography/Title";
```
Comment (gabrielsaruhashi) [inline src/components/Profile/ProfileInfo.tsx:5]: Same here.

I'd do Typography.Title
Typography.Text

File: src/components/Profile/ProfileInfo.tsx [hunk 17]
```diff
@@ -0,0 +1,111 @@
...
+
+interface Props {
+  user: User;
+  calls: BaseCall[];
+  onEdit: () => void;
+}
+
+const ProfileInfo: React.FC<Props> = ({ user, calls, onEdit }) => {
+  const { t } = useTranslation("profile");
+  const totalMinutes = calls
```
Comment (gabrielsaruhashi) [inline src/components/Profile/ProfileInfo.tsx:23]: I'd move this to a useState useEffect so we don't need to re-calculate it on every re-render

**Review by gabrielsaruhashi — CHANGES_REQUESTED** [review]
final round of feedback 🔥

File: src/App.tsx (hunk 10 above)
Comment (lspdrz) [reply src/App.tsx:24]: YEs it did!!

File: src/components/menu/Sidebar.tsx (hunk 13 above)
Comment (lspdrz) [reply src/components/menu/Sidebar.tsx:35]: @gabrielsaruhashi I think that Liz used the Drawer component (which automatically comes with box shadow) in her design. But we want to use Sider here, which is why I'm manually setting the box shadow. I can change it to tailwind though

Comment (gabrielsaruhashi) [reply src/components/menu/Sidebar.tsx:35]: oh I see

Ye! Would be good to use tailwind's then!

File: src/components/menu/Sidebar.tsx [hunk 18]
```diff
@@ -1,51 +1,76 @@
...
-        {/* <Menu.Item
-          key="contacts"
-          icon={<UserOutlined />}
-          onClick={() => navigate("home")}
+    <Sider theme="light" className="shadow-lg">
+      <Space direction="vertical" className="w-full">
+        <Row className="pt-2">
+          <Logo className="login-logo" />
+        </Row>
+        <Row></Row>
```
Comment (gabrielsaruhashi) [inline src/components/menu/Sidebar.tsx:38]: delete?

**Review by gabrielsaruhashi — APPROVED** [review]
stamped! + one nit

**Pull request merged**